p2p/src/sync/types.rs

```diff
@@ -17,17 +17,40 @@ use std::time::Duration;
 
 /// Activity with a peer.
 #[derive(Debug)]
-pub enum PeerActivity {
-    /// Node is pending for further actions with a peer.
-    Pending,
-    /// Node has sent a header list request to a peer and is expecting a header list response.
-    ExpectingHeaderList {
-        /// A time when the header list request was sent.
-        time: Duration,
-    },
-    /// Node has sent a block list request to a peer and is expecting block responses.
-    ExpectingBlocks {
-        /// A time when either the block list request was sent or last block response was received.
-        time: Duration,
-    },
+pub struct PeerActivity {
+    expecting_headers_since: Option<Duration>,
```

Comment:
Nitpick: "since" should be used with a reference point in time, not with a distance in time. So we say "I've been doing that since 2020" or "I've been doing that for 3 years" or "I started doing that 3 years ago". Whichever is the implication. Now feel free to correct my English :smile:. I believe it's a common mistake.

Reply:
Never mind. I realized this is a timestamp written as duration... and this is why types are important :D